engines/stark/services/userinterface.h

```diff
@@ -96,7 +96,9 @@ class UserInterface {
 	void backPrevScreen();
 
 	/** Back to the main menu screen and rest resources */
-	void quitToMainMenu();
+	void requestQuitToMainMenu() { _quitToMainMenu = true; }
+	bool hasQuitToMainMenuRequest() { return _quitToMainMenu; }
+	void performQuitToMainMenu();
```

Comment:
`Script` needs to request quitting to the main menu. In the previous implementation, immediately quitting will trigger the deallocation of the `Script` object and therefore crash the game ~~and it took me an eternity to realize this!!!~~ . So it is cached here and is performed later in the update part of the game's main loop.

Reply:
Building with Address Sanitizer enabled helps a lot to understand these issues. It tells precisely the cause of the use after free. It's probably a bit of a pain to get it working on Windows, and IMO it's well worth having a Linux install somewhere just to be able to use it when needed.

Reply:
Yeah now I know the good of Linux...